lib/muffin_man/product_pricing/v0.rb

```diff
@@ -6,14 +6,20 @@ class V0 < SpApiClient
       GET_COMPETITIVE_PRICE_PARAMS = %w[Asins Skus CustomerType].freeze
 
       def get_competitive_pricing(marketplace_id, item_type='Asin', params = {})
-        @params = params
-        @local_var_path = "/products/pricing/v0/competitivePrice"
-        @query_params = { "MarketplaceId" => marketplace_id,
-                          "ItemType" => item_type }
-        @query_params.merge!(@params.slice(*GET_COMPETITIVE_PRICE_PARAMS))
+        @local_var_path = '/products/pricing/v0/competitivePrice'
+        @params = { 'MarketplaceId' => marketplace_id, 'ItemType' => item_type }
+        @params.merge! params.slice(*GET_COMPETITIVE_PRICE_PARAMS)
+        @query_params = hash_to_sp_api_uri_params(@params)
         @request_type = "GET"
         call_api
       end
+
+      private
+
+      # SP APIs expect param array on the form of Asins=Asin1%2CAsin2 (%2C is url escaped for ,) ...
+      def hash_to_sp_api_uri_params(params)
+        params.keys.map { |k| "#{k}=#{params[k].is_a?(Array) ? params[k].join("%2C") : params[k]}" }.join('&')
```

Comment:
@gavinritchie, Great questions and you are correct. Typhoeus does encode any url params. However, When I first tried the "," instead of "%2C" (escaped comma) I got "code": "InvalidSignature" (I believe we are not encoding url params when calculating the signature).
On your second point, you are also correct. Typhoeus will map a hash with arrays to a string. However, if I pass: 
hash = {'Asins' => ['a1','a2']}
Typhoeus will covert to 'Asins%5B0%5D=a1&Asins%5B1%5D=a2' (decoded 'Asins[0]=a1&Asins[1]=a2') and SP API expects Asins=a1%2Ca2 (decoded 'Asins=a1,a2')

Reply:
Got it, thanks for the explanation!